gen/flights/flightplan.py

```diff
@@ -1182,14 +1182,24 @@ def racetrack_for_objective(
         start = end.point_from_heading(heading - 180, diameter)
         return start, end
 
-    @staticmethod
-    def aewc_orbit(location: MissionTarget) -> Point:
+    def aewc_orbit(self, location: MissionTarget) -> Point:
         closest_airfield = location
-        # TODO: This is a heading to itself.
-        # Place this either over the target or as close as possible outside the
-        # threat zone: https://github.com/Khopa/dcs_liberation/issues/842.
         heading = location.position.heading_between_point(closest_airfield.position)
-        return location.position.point_from_heading(heading, 5000)
+
+        position = ShapelyPoint(
+            self.package.target.position.x, self.package.target.position.y
+        )
+
+        if meters(position.distance(self.threat_zones.all)) == meters(0):
+            distance_to_no_fly = (
+                    meters(position.distance(self.threat_zones.all))
+                    - self.doctrine.cap_engagement_range
```

Comment:
This distance isn't meaningful here. The reason it was used for the BARCAP planning is that its goal is to keep the edge of the engagement zone out of the enemy threat zone. Here we just need to keep the orbit itself safe. Instead, just pick a larger constant than 5nm. 40nm seems reasonable.